subsys/fb/cfb_shell.c

```diff
@@ -0,0 +1,355 @@
+	} else {
+		shell_error(shell, "wrong parameter name");
+		shell_help(shell);
+		return 1;
+	}
+
+	col = strtol(argv[2], NULL, 10);
+	if (col > cfb_get_display_parameter(dev, CFB_DISPLAY_COLS)) {
+		shell_error(shell, "Invalid col=%d position", col);
+		return -EINVAL;
+	}
+
+	row = strtol(argv[3], NULL, 10);
+	if (row > cfb_get_display_parameter(dev, CFB_DISPLAY_ROWS)) {
+		shell_error(shell, "Invalid row=%d position", row);
+		return -EINVAL;
+	}
+
+	if (scroll) {
+		boundary -= row;
+	} else {
+		col++;
+		boundary -= col;
+	}
+
+	for (int i = 0; i < boundary; i++) {
+		err = cmd_cfb_print(shell, col, row, argv[4]);
+		if (err) {
+			shell_error(shell,
+				    "Failed printing to Framebuffer error=%d",
+				    err);
+		}
+		if (scroll) {
+			row++;
+		} else {
+			col++;
+		}
+	}
+
+	cmd_cfb_print(shell, 0, 0, "");
+
+	return err;
+}
+
+static int cmd_set_font(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err;
+	int idx;
+	u8_t height;
+	u8_t width;
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	idx = strtol(argv[1], NULL, 10);
+
+	err = cfb_get_font_size(dev, idx, &width, &height);
+	if (err) {
+		shell_error(shell, "Invalid font idx=%d err=%d\n", idx, err);
+		return err;
+	}
+
+	err = cfb_framebuffer_set_font(dev, idx);
+	if (err) {
+		shell_error(shell, "Failed setting font idx=%d err=%d", idx,
+			    err);
+		return err;
+	}
+
+	shell_print(shell, "Font idx=%d height=%d widht=%d set", idx, height,
+		    width);
+
+	return err;
+}
+
+static int cmd_invert(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err;
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	err = cfb_framebuffer_invert(dev);
+	if (err) {
+		shell_error(shell, "Error inverting Framebuffer");
+		return err;
+	}
+
+	cmd_cfb_print(shell, 0, 0, "");
+
+	shell_print(shell, "Framebuffer Inverted");
+
+	return err;
+}
+
+static int cmd_get_fonts(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err = 0;
+	u8_t font_height;
+	u8_t font_width;
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	for (int idx = 0; idx < cfb_get_numof_fonts(dev); idx++) {
+		if (cfb_get_font_size(dev, idx, &font_width, &font_height)) {
+			break;
+		}
+		shell_print(shell, "idx=%d height=%d width=%d", idx,
+			    font_height, font_width);
+	}
+
+	return err;
+}
+
+static int cmd_get_device(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err = 0;
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	shell_print(shell, "Framebuffer Device: %s", DISPLAY_DRIVER);
+
+	return err;
+}
+
+static int cmd_get_param(const struct shell *shell, size_t argc, char *argv[])
+{
+	int param = 0;
+	static const char * const param_name[] = {
+		"height", "width", "ppt", "rows", "cols"};
+	const char *param_filter = argv[1];
+
+	if (!dev) {
+		shell_error(shell, HELP_INIT);
+		return -ENODEV;
+	}
+
+	if (!strcmp(param_filter, "all")) {
```

Comment:
Similar situation. These all can be subcommands with a handler - you won't need `strcmp` and `cmd_get_param` or
these can be subcommands without a handler what will simplify shell usage.